python/cuml/manifold/umap.pyx

```diff
@@ -437,7 +445,8 @@ class UMAP(Base):
         elif isinstance(X, cupy.ndarray):
             return cupy.array(embedding)
 
-    def fit(self, X, y=None, convert_dtype=True):
+    def fit(self, X, y=None, convert_dtype=True,
```

Comment:
Rather than accepting two separate arrays for knn indices and knn dists, I'm wondering if we should just accept a single Scipy / CuPy sparse CSR / COO matrix where the row indices are a simple `cp.arange` with a skip of size `n_neighbors`, the `knn_indices` are the columns, and the `knn_dists` are the values. I think this will be much more clean and easier to manage once we end up accepting the knn for continuous labels. This should also be a trivial change, since sparse matrices in scipy and cupy are going to be backed by either numpy or cupy arrays. 

We should also throw an exception in the case the user is doing a (semi-)supervised training w/ discrete labels (target_metric == categorical) and passing a knn graph as input.